erpnext/stock/doctype/stock_entry/test_stock_entry.py

```
@@ -2250,6 +2250,145 @@ def test_transferred_qty_in_material_transfer(self):
+				"has_batch_no": 1,
+				"has_serial_no": 1,
+				"create_new_batch": 1,
+				"batch_number_series": "SAMPLE-RET-.#####",
+				"serial_no_series": "SAMPLE-RET-SN-.#####",
+			},
+		)
+		material_receipt = make_stock_entry(
+			item_code=retain_sample_item.item_code, target=warehouse, qty=10, purpose="Material Receipt"
+		)
+
+		source_sabb = frappe.get_doc(
```

Ruff (0.14.14)

`frappe` may be undefined, or defined from star imports

(F405)